jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java

```diff
@@ -138,7 +138,17 @@ private String makeUserAgent() {
    * @return the new {@link Factory}
    */
   public static Factory factory(EventHandlers eventHandlers, String serverUrl, String imageName) {
-    return new Factory(eventHandlers, new RegistryEndpointRequestProperties(serverUrl, imageName));
+    return factory(eventHandlers, serverUrl, imageName, null);
+  }
+
+  public static Factory factory(
+      EventHandlers eventHandlers,
+      String serverUrl,
+      String imageName,
+      @Nullable String sourceImageName) {
```

Comment:
Is there a situation where this is null and the [previous `factory(` method](https://github.com/GoogleContainerTools/jib/pull/1793/files#diff-3482bc5337be75b1d8e501d4ee8a9654R140) would not be sufficient?

Reply:
oicwyga — I initially was trying to get rid of the 2-arg version of `RegistryEndpointRequestProperties`.  Will restore.

Reply:
and then I can remove the nullable